### PR TITLE
Feature/calendar-get-sum-of-working-hours

### PR DESCRIPTION
**Pull request by d-mashchenko**
- show required number of woking hours for month (with holidays and vacations)
- add icon with day off description near date cell in calendar

### CODE REVIEW

**Review by o-mehlii — COMMENTED** [review]
Only the first day of vacation is displayed in the calendar.

File: renderer/components/Calendar/Calendar.tsx [hunk 1]
```diff
@@ -47,48 +49,59 @@ type CalendarProps = {
   setSelectedDate: Dispatch<SetStateAction<Date>>;
```
Comment (a-nahaitsev) [inline renderer/components/Calendar/Calendar.tsx:49]: let's rewrite name of constant with uppercase
https://www.javascripttutorial.net/es6/javascript-const/

File: renderer/components/Calendar/Calendar.tsx [hunk 2]
```diff
@@ -139,13 +152,96 @@ export function Calendar({
...
+      );
+
+      const userEmail = timetrackerUserInfo?.email;
+      const vacationsPromise = global.ipcRenderer.invoke(
+        "timetracker:get-vacations",
+        plannerToken,
+        userEmail
+      );
+
+      userPromises.push(...[holidaysPromise, vacationsPromise]);
```
Comment (a-nahaitsev) [inline renderer/components/Calendar/Calendar.tsx:183]: if i'm not mistaken you can write it like this `userPromises.push(holidaysPromise, vacationsPromise);`

File: renderer/components/Calendar/Calendar.tsx [hunk 3]
```diff
@@ -139,13 +152,96 @@ export function Calendar({
...
+        });
+      });
+
+      vacationsAndSickdays.forEach((item) => {
+        userDaysOff.push({
+          date: new Date(item?.dateFrom),
+          duration: item?.quantity,
+          description: item?.description,
+        });
+      });
```
Comment (a-nahaitsev) [inline renderer/components/Calendar/Calendar.tsx:226]: i would move it to another function
i would return data from `loadHolidaysAndVacations` and moved this function to utils, file like timetracker.ts and import it here
what will you say?

**Review by a-nahaitsev — CHANGES_REQUESTED** [review]

**Comment by a-nahaitsev** [comment]
Forgot to say
Can we load already reported hours from API and add current report hours until you load it?
Now you are calculating it from reports. But if one of them will be removed? or you changed your device and you have no reports for previous days. And also for previous months i see 0 in `Total`

**Review by a-nahaitsev — APPROVED** [review]

**Pull request merged**